### PR TITLE
Fix NPE at function wrapper call

### LINKED ISSUE #25946: Getting NPE when compiling an HTTP service with service/resource annotations
**Description:**
During the debug it was found that NPE occurs at https://github.com/ballerina-platform/ballerina-lang/blob/bd89cf09e6428fe9397090af071acb5ef0b1fa74/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java#L1502 due to `funcSymbol` becoming Null. Further `lookupBIRFunctionWrapper` function looks for a wrapper with a key contains `funcName`  Eg: `$annot_func$2103`. But wrapper has been added with a slightly different key Eg: `$annot_func$#2103`. So `BIRFunctionWrapper` becomes null.

Thank you @warunalakshitha and @MaryamZi for the help given to find out the issue

**Steps to reproduce:**
The source is available in a brach(https://github.com/ballerina-platform/module-ballerina-http/tree/native) of HTTP repo.
Clone it and run `./gradlew test`

Internal log
```
[2020-09-18 20:01:00,123] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generateModuleClasses(JvmPackageGen.java:446)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:434)
	at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:157)
	at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.lambda$generate$0(CodeGenerator.java:139)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:133)
	at org.wso2.ballerinalang.compiler.CompilerDriver.codeGen(CompilerDriver.java:317)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compile(CompilerDriver.java:313)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackageSymbol(CompilerDriver.java:250)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackage(CompilerDriver.java:142)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:176)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:147)
	at org.ballerinalang.packerina.task.CompileTask.execute(CompileTask.java:69)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.TestCommand.execute(TestCommand.java:434)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:57)
Caused by: java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmMethodGen.generateLambdaMethod(JvmMethodGen.java:1502)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.lambda$generateModuleClasses$0(JvmPackageGen.java:501)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1696)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
 

```

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose

Fixes #25946 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by niveathika** [comment]
@chamil321 Lets check http build locally with this fix and check

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java [hunk 1]
```diff
@@ -1491,7 +1492,8 @@ void generateLambdaMethod(BIRInstruction ins, ClassWriter cw, String lambdaName)
             mv.visitMethodInsn(INVOKEINTERFACE, OBJECT_VALUE, "call", methodDesc, true);
         } else {
             String jvmClass;
-            String lookupKey = JvmCodeGenUtil.getPackageName(orgName, moduleName, version) + funcName;
+            String encodedFuncName = IdentifierEncoder.encodeIdentifier(funcName);
+            String lookupKey = JvmCodeGenUtil.getPackageName(orgName, moduleName, version) + encodedFuncName;
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java:1496]: @HindujaB Shall we write a test case for this fix? May be as a separate PR if we need to merge this quickly.

**Comment by chamil321** [comment]
> @chamil321 Lets check http build locally with this fix and check

Tested HTTP module locally, the issue has fixed.

**Comment by niveathika** [comment]
@warunalakshitha As this is a blocker for the release this week, shall we merge and @HindujaB can send the test case in separate PR as you suggested.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java (hunk 1 above)
Comment (HindujaB) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java:1496]: I'll send a separate PR for the test

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**